{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "3f2e6290",
   "metadata": {},
   "source": [
    "# Data Preparation"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d85477cf-b3ba-4aa5-b41b-bf8299be47a5",
   "metadata": {},
   "source": [
    "This notebook prepares data on allele frequency trajectories of two *SLC22A10* proxy variants for rs1790218 at chr11: 63,057,925 (hg19). First, load the Pandas package and set viewing options."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c084c9a4",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "pd.set_option('display.max_columns', 20)\n",
    "pd.set_option('display.max_rows', 500)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2763e6f2",
   "metadata": {},
   "source": [
    "Change directories."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "bf4fa44a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/wynton/group/capra/projects/SLC22A10/data\n"
     ]
    }
   ],
   "source": [
    "cd /wynton/group/capra/projects/SLC22A10/data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "28b0fd58",
   "metadata": {},
   "source": [
    "Let's start with the sample annotation file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "9e719583",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genetic ID</th>\n",
       "      <th>Master ID</th>\n",
       "      <th>Skeletal code</th>\n",
       "      <th>Skeletal element</th>\n",
       "      <th>Year data from this individual was first published [for a present-day individuals we give the data of the data reported here; missing GreenScience 2010 (Vi33.15, Vi33.26), Olalde2018 (I2657), RasmussenNature2010 (Australian)]</th>\n",
       "      <th>Publication</th>\n",
       "      <th>Method for Determining Date; unless otherwise specified, calibrations use 95.4% intervals from OxCal v4.4.2 Bronk Ramsey (2009); r5; Atmospheric data from Reimer et al (2020)</th>\n",
       "      <th>Date mean in BP in years before 1950 CE [OxCal mu for a direct radiocarbon date, and average of range for a contextual date]</th>\n",
       "      <th>Date standard deviation in BP [OxCal sigma for a direct radiocarbon date, and standard deviation of the uniform distribution between the two bounds for a contextual date]</th>\n",
       "      <th>Full Date One of two formats. (Format 1) 95.4% CI calibrated radiocarbon age (Conventional Radiocarbon Age BP, Lab number) e.g. 2624-2350 calBCE (3990±40 BP, Ua-35016). (Format 2) Archaeological context range, e.g. 2500-1700 BCE</th>\n",
       "      <th>...</th>\n",
       "      <th>Y haplogroup (manual curation in ISOGG format)</th>\n",
       "      <th>mtDNA coverage (merged data)</th>\n",
       "      <th>mtDNA haplogroup if &gt;2x or published</th>\n",
       "      <th>mtDNA match to consensus if &gt;2x (merged data)</th>\n",
       "      <th>Damage rate in first nucleotide on sequences overlapping 1240k targets (merged data)</th>\n",
       "      <th>Sex ratio [Y/(Y+X) counts] (merged data)</th>\n",
       "      <th>Library type (minus=no.damage.correction, half=damage.retained.at.last.position, plus=damage.fully.corrected, ds=double.stranded.library.preparation, ss=single.stranded.library.preparation)</th>\n",
       "      <th>Libraries</th>\n",
       "      <th>ASSESSMENT</th>\n",
       "      <th>ASSESSMENT WARNINGS (Xcontam interval is listed if lower bound is &gt;0.005, \"QUESTIONABLE\" if lower bound is 0.01-0.02, \"QUESTIONABLE_CRITICAL\" or \"FAIL\" if lower bound is &gt;0.02) (mtcontam confidence interval is listed if coverage &gt;2 and upper bound is &lt;0.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Ne30_genotyping_noUDG</td>\n",
       "      <td>NE30</td>\n",
       "      <td>AR9.9K_2d.rel.NE-4_deleted</td>\n",
       "      <td>..</td>\n",
       "      <td>2021</td>\n",
       "      <td>MaoFuCell2021</td>\n",
       "      <td>Direct IntCal20</td>\n",
       "      <td>9896</td>\n",
       "      <td>121</td>\n",
       "      <td>8175-7750 calBCE (8825±30 BP, BA-152174)</td>\n",
       "      <td>...</td>\n",
       "      <td>C</td>\n",
       "      <td>99</td>\n",
       "      <td>D4m</td>\n",
       "      <td>..</td>\n",
       "      <td>12</td>\n",
       "      <td>..</td>\n",
       "      <td>ss.minus</td>\n",
       "      <td>HRR163270</td>\n",
       "      <td>PASS</td>\n",
       "      <td>..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Ne61_genotyping_noUDG</td>\n",
       "      <td>NE61</td>\n",
       "      <td>AR3.4K_LowCov</td>\n",
       "      <td>..</td>\n",
       "      <td>2021</td>\n",
       "      <td>MaoFuCell2021</td>\n",
       "      <td>Direct: IntCal20</td>\n",
       "      <td>3421</td>\n",
       "      <td>23</td>\n",
       "      <td>1510-1425 calBCE (3210±25 BP, BA-152205)</td>\n",
       "      <td>...</td>\n",
       "      <td>n/a (Sex Unknown)</td>\n",
       "      <td>574</td>\n",
       "      <td>D4b1a2a</td>\n",
       "      <td>..</td>\n",
       "      <td>5</td>\n",
       "      <td>..</td>\n",
       "      <td>ss.minus</td>\n",
       "      <td>HRR163284</td>\n",
       "      <td>PASS</td>\n",
       "      <td>..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Ne35_genotyping_noUDG</td>\n",
       "      <td>NE35</td>\n",
       "      <td>AR8.9K</td>\n",
       "      <td>..</td>\n",
       "      <td>2021</td>\n",
       "      <td>MaoFuCell2021</td>\n",
       "      <td>Direct: IntCal20</td>\n",
       "      <td>8990</td>\n",
       "      <td>72</td>\n",
       "      <td>7172-6831 calBCE (8075±30 BP, BA-152179)</td>\n",
       "      <td>...</td>\n",
       "      <td>n/a (female)</td>\n",
       "      <td>602</td>\n",
       "      <td>R11</td>\n",
       "      <td>..</td>\n",
       "      <td>22</td>\n",
       "      <td>..</td>\n",
       "      <td>ss.minus</td>\n",
       "      <td>HRR163273</td>\n",
       "      <td>PASS</td>\n",
       "      <td>..</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>I17622</td>\n",
       "      <td>I17622</td>\n",
       "      <td>2637; Tumulus 1, grave ?, 10</td>\n",
       "      <td>tooth (molar)</td>\n",
       "      <td>2022</td>\n",
       "      <td>LazaridisAlpaslanRoodenbergScience2022</td>\n",
       "      <td>Context: Main period of tumulus use</td>\n",
       "      <td>3050</td>\n",
       "      <td>173</td>\n",
       "      <td>1400-800 BCE</td>\n",
       "      <td>...</td>\n",
       "      <td>..</td>\n",
       "      <td>0.5948</td>\n",
       "      <td>n/a (&lt;2x)</td>\n",
       "      <td>n/a (&lt;2x)</td>\n",
       "      <td>0.096</td>\n",
       "      <td>0.375</td>\n",
       "      <td>ss.half</td>\n",
       "      <td>S17622.E1.L1</td>\n",
       "      <td>PASS</td>\n",
       "      <td>2500.to.5000.SNPs, damage.ss.half=0.096</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>I13833</td>\n",
       "      <td>I13833</td>\n",
       "      <td>1234; Tumulus 2, grave 3, ind. 1, 2</td>\n",
       "      <td>petrous</td>\n",
       "      <td>2022</td>\n",
       "      <td>LazaridisAlpaslanRoodenbergScience2022</td>\n",
       "      <td>Context: Date(s) on individuals with &gt;20cM IBD...</td>\n",
       "      <td>325</td>\n",
       "      <td>101</td>\n",
       "      <td>1450-1800 CE</td>\n",
       "      <td>...</td>\n",
       "      <td>..</td>\n",
       "      <td>1150.387048</td>\n",
       "      <td>X2e1</td>\n",
       "      <td>[0.985,0.997]</td>\n",
       "      <td>0.122</td>\n",
       "      <td>0.435054726</td>\n",
       "      <td>ds.half,ss.half,ss.half</td>\n",
       "      <td>S13833.Y1.E1.L1,S20786.Y1.E1.L1,S20790.Y1.E1.L1</td>\n",
       "      <td>PASS</td>\n",
       "      <td>..</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Genetic ID Master ID                        Skeletal code  \\\n",
       "0  Ne30_genotyping_noUDG      NE30           AR9.9K_2d.rel.NE-4_deleted   \n",
       "1  Ne61_genotyping_noUDG      NE61                        AR3.4K_LowCov   \n",
       "2  Ne35_genotyping_noUDG      NE35                               AR8.9K   \n",
       "3                 I17622    I17622         2637; Tumulus 1, grave ?, 10   \n",
       "4                 I13833    I13833  1234; Tumulus 2, grave 3, ind. 1, 2   \n",
       "\n",
       "  Skeletal element  \\\n",
       "0               ..   \n",
       "1               ..   \n",
       "2               ..   \n",
       "3    tooth (molar)   \n",
       "4          petrous   \n",
       "\n",
       "  Year data from this individual was first published [for a present-day individuals we give the data of the data reported here; missing GreenScience 2010 (Vi33.15, Vi33.26), Olalde2018 (I2657), RasmussenNature2010 (Australian)]  \\\n",
       "0                                               2021                                                                                                                                                                                  \n",
       "1                                               2021                                                                                                                                                                                  \n",
       "2                                               2021                                                                                                                                                                                  \n",
       "3                                               2022                                                                                                                                                                                  \n",
       "4                                               2022                                                                                                                                                                                  \n",
       "\n",
       "                              Publication  \\\n",
       "0                           MaoFuCell2021   \n",
       "1                           MaoFuCell2021   \n",
       "2                           MaoFuCell2021   \n",
       "3  LazaridisAlpaslanRoodenbergScience2022   \n",
       "4  LazaridisAlpaslanRoodenbergScience2022   \n",
       "\n",
       "  Method for Determining Date; unless otherwise specified, calibrations use 95.4% intervals from OxCal v4.4.2 Bronk Ramsey (2009); r5; Atmospheric data from Reimer et al (2020)  \\\n",
       "0                                    Direct IntCal20                                                                                                                               \n",
       "1                                   Direct: IntCal20                                                                                                                               \n",
       "2                                   Direct: IntCal20                                                                                                                               \n",
       "3                Context: Main period of tumulus use                                                                                                                               \n",
       "4  Context: Date(s) on individuals with >20cM IBD...                                                                                                                               \n",
       "\n",
       "   Date mean in BP in years before 1950 CE [OxCal mu for a direct radiocarbon date, and average of range for a contextual date]  \\\n",
       "0                                               9896                                                                              \n",
       "1                                               3421                                                                              \n",
       "2                                               8990                                                                              \n",
       "3                                               3050                                                                              \n",
       "4                                                325                                                                              \n",
       "\n",
       "   Date standard deviation in BP [OxCal sigma for a direct radiocarbon date, and standard deviation of the uniform distribution between the two bounds for a contextual date]  \\\n",
       "0                                                121                                                                                                                            \n",
       "1                                                 23                                                                                                                            \n",
       "2                                                 72                                                                                                                            \n",
       "3                                                173                                                                                                                            \n",
       "4                                                101                                                                                                                            \n",
       "\n",
       "  Full Date One of two formats. (Format 1) 95.4% CI calibrated radiocarbon age (Conventional Radiocarbon Age BP, Lab number) e.g. 2624-2350 calBCE (3990±40 BP, Ua-35016). (Format 2) Archaeological context range, e.g. 2500-1700 BCE  \\\n",
       "0           8175-7750 calBCE (8825±30 BP, BA-152174)                                                                                                                                                                                     \n",
       "1           1510-1425 calBCE (3210±25 BP, BA-152205)                                                                                                                                                                                     \n",
       "2           7172-6831 calBCE (8075±30 BP, BA-152179)                                                                                                                                                                                     \n",
       "3                                       1400-800 BCE                                                                                                                                                                                     \n",
       "4                                       1450-1800 CE                                                                                                                                                                                     \n",
       "\n",
       "   ... Y haplogroup (manual curation in ISOGG format)  \\\n",
       "0  ...                                              C   \n",
       "1  ...                              n/a (Sex Unknown)   \n",
       "2  ...                                   n/a (female)   \n",
       "3  ...                                             ..   \n",
       "4  ...                                             ..   \n",
       "\n",
       "  mtDNA coverage (merged data) mtDNA haplogroup if >2x or published  \\\n",
       "0                           99                                  D4m   \n",
       "1                          574                              D4b1a2a   \n",
       "2                          602                                  R11   \n",
       "3                       0.5948                            n/a (<2x)   \n",
       "4                  1150.387048                                 X2e1   \n",
       "\n",
       "  mtDNA match to consensus if >2x (merged data)  \\\n",
       "0                                            ..   \n",
       "1                                            ..   \n",
       "2                                            ..   \n",
       "3                                     n/a (<2x)   \n",
       "4                                 [0.985,0.997]   \n",
       "\n",
       "  Damage rate in first nucleotide on sequences overlapping 1240k targets (merged data)  \\\n",
       "0                                                 12                                     \n",
       "1                                                  5                                     \n",
       "2                                                 22                                     \n",
       "3                                              0.096                                     \n",
       "4                                              0.122                                     \n",
       "\n",
       "  Sex ratio [Y/(Y+X) counts] (merged data)  \\\n",
       "0                                       ..   \n",
       "1                                       ..   \n",
       "2                                       ..   \n",
       "3                                    0.375   \n",
       "4                              0.435054726   \n",
       "\n",
       "  Library type (minus=no.damage.correction, half=damage.retained.at.last.position, plus=damage.fully.corrected, ds=double.stranded.library.preparation, ss=single.stranded.library.preparation)  \\\n",
       "0                                           ss.minus                                                                                                                                              \n",
       "1                                           ss.minus                                                                                                                                              \n",
       "2                                           ss.minus                                                                                                                                              \n",
       "3                                            ss.half                                                                                                                                              \n",
       "4                            ds.half,ss.half,ss.half                                                                                                                                              \n",
       "\n",
       "                                         Libraries ASSESSMENT  \\\n",
       "0                                        HRR163270       PASS   \n",
       "1                                        HRR163284       PASS   \n",
       "2                                        HRR163273       PASS   \n",
       "3                                     S17622.E1.L1       PASS   \n",
       "4  S13833.Y1.E1.L1,S20786.Y1.E1.L1,S20790.Y1.E1.L1       PASS   \n",
       "\n",
       "  ASSESSMENT WARNINGS (Xcontam interval is listed if lower bound is >0.005, \"QUESTIONABLE\" if lower bound is 0.01-0.02, \"QUESTIONABLE_CRITICAL\" or \"FAIL\" if lower bound is >0.02) (mtcontam confidence interval is listed if coverage >2 and upper bound is <0.  \n",
       "0                                                 ..                                                                                                                                                                                                              \n",
       "1                                                 ..                                                                                                                                                                                                              \n",
       "2                                                 ..                                                                                                                                                                                                              \n",
       "3            2500.to.5000.SNPs, damage.ss.half=0.096                                                                                                                                                                                                              \n",
       "4                                                 ..                                                                                                                                                                                                              \n",
       "\n",
       "[5 rows x 35 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_annotation = pd.read_csv('/wynton/group/capra/data/wynton_databases/ancient_dna/V54.1.p1/v54.1.p1_1240K_public.anno', sep = '\\t', header = 0, low_memory = False)\n",
    "sample_annotation.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "4ca36f3d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "16388"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(sample_annotation['Genetic ID'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "36e9d53f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "13357"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(sample_annotation['Master ID'].unique())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "18d4a585",
   "metadata": {},
   "source": [
    "Let's create a dictionary of genetic IDs and master IDs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "30320bb6",
   "metadata": {},
   "outputs": [],
   "source": [
    "IDs_dict = dict(zip(sample_annotation['Genetic ID'], sample_annotation['Master ID']))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b43347a1",
   "metadata": {},
   "source": [
    "Build a dictionary of sample dates and check the dictionary."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "e60fded1",
   "metadata": {},
   "outputs": [],
   "source": [
    "dates_dict = dict(zip(sample_annotation['Genetic ID'], sample_annotation['Date mean in BP in years before 1950 CE [OxCal mu for a direct radiocarbon date, and average of range for a contextual date]']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "623f396c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "40022"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dates_dict['Oase1_noUDG']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e0061d9d",
   "metadata": {},
   "source": [
    "Now let's get sample locations. We need to look in a different file so read that in and build a dictionary again."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "8e7c03e4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>sex</th>\n",
       "      <th>location</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Ne30_genotyping_noUDG</td>\n",
       "      <td>M</td>\n",
       "      <td>China_AmurRiver_EarlyN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Ne61_genotyping_noUDG</td>\n",
       "      <td>U</td>\n",
       "      <td>China_AmurRiver_BA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Ne35_genotyping_noUDG</td>\n",
       "      <td>F</td>\n",
       "      <td>China_AmurRiver_EarlyN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>I13833</td>\n",
       "      <td>M</td>\n",
       "      <td>Albania_EarlyModern_oCaucasus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>I3931</td>\n",
       "      <td>M</td>\n",
       "      <td>Armenia_Aknashen_N</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  sample sex                       location\n",
       "0  Ne30_genotyping_noUDG   M         China_AmurRiver_EarlyN\n",
       "1  Ne61_genotyping_noUDG   U             China_AmurRiver_BA\n",
       "2  Ne35_genotyping_noUDG   F         China_AmurRiver_EarlyN\n",
       "3                 I13833   M  Albania_EarlyModern_oCaucasus\n",
       "4                  I3931   M             Armenia_Aknashen_N"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_sex_location_header = ['sample', 'sex', 'location']\n",
    "sample_sex_location = pd.read_csv('/wynton/group/capra/projects/SLC22A10/data/rs1790218_proxy_variants.ind', delim_whitespace = True, names = sample_sex_location_header)\n",
    "sample_sex_location.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "5bac7011",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "13017"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(sample_sex_location)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "ce49826d",
   "metadata": {},
   "outputs": [],
   "source": [
    "location_dict = dict(zip(sample_sex_location['sample'], sample_sex_location['location']))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0eccc072-c9df-4231-ae33-fd46c67d27b0",
   "metadata": {},
   "source": [
    "Test the dictionary."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "d4408a8e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Romania_Oase_UP_enhanced'"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "location_dict['Oase1_noUDG']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fdd603e0",
   "metadata": {},
   "source": [
    "Let's also map on higher-level geographic categories from a text file. These were manually curated by myself based on the sampling locality country."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "8328fd4d",
   "metadata": {},
   "outputs": [],
   "source": [
    "subregion_dict = {}\n",
    "region_dict = {}\n",
    "\n",
    "with open('/wynton/group/capra/projects/SLC22A10/data/AADR_sample_subregion_region_mapping.txt', 'r') as file:\n",
    "    for line in file:\n",
    "        sample, subregion, region = line.strip().split('\\t')  # Assumes tab-separated columns, adjust delimiter as needed\n",
    "        subregion_dict[sample] = subregion\n",
    "        region_dict[sample] = region"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "61a0b3a5",
   "metadata": {},
   "source": [
    "Now read in the genotypes. We will exclude a third variant pulled from the AADR, an Affymetrix locus without an rsID. We will also map on the sample metadata stored in dictionaries above."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "f3b96932",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsID</th>\n",
       "      <th>sample_ID</th>\n",
       "      <th>sample</th>\n",
       "      <th>date</th>\n",
       "      <th>location</th>\n",
       "      <th>subregion</th>\n",
       "      <th>region</th>\n",
       "      <th>gt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>NE30</td>\n",
       "      <td>Ne30_genotyping_noUDG</td>\n",
       "      <td>9896.0</td>\n",
       "      <td>China_AmurRiver_EarlyN</td>\n",
       "      <td>East Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>NE61</td>\n",
       "      <td>Ne61_genotyping_noUDG</td>\n",
       "      <td>3421.0</td>\n",
       "      <td>China_AmurRiver_BA</td>\n",
       "      <td>East Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>NE35</td>\n",
       "      <td>Ne35_genotyping_noUDG</td>\n",
       "      <td>8990.0</td>\n",
       "      <td>China_AmurRiver_EarlyN</td>\n",
       "      <td>East Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I13833</td>\n",
       "      <td>I13833</td>\n",
       "      <td>325.0</td>\n",
       "      <td>Albania_EarlyModern_oCaucasus</td>\n",
       "      <td>Eastern Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I3931</td>\n",
       "      <td>I3931</td>\n",
       "      <td>7858.0</td>\n",
       "      <td>Armenia_Aknashen_N</td>\n",
       "      <td>Central Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        rsID sample_ID                 sample    date  \\\n",
       "0  rs1783634      NE30  Ne30_genotyping_noUDG  9896.0   \n",
       "1  rs1783634      NE61  Ne61_genotyping_noUDG  3421.0   \n",
       "2  rs1783634      NE35  Ne35_genotyping_noUDG  8990.0   \n",
       "3  rs1783634    I13833                 I13833   325.0   \n",
       "4  rs1783634     I3931                  I3931  7858.0   \n",
       "\n",
       "                        location       subregion  region  gt  \n",
       "0         China_AmurRiver_EarlyN       East Asia    Asia   1  \n",
       "1             China_AmurRiver_BA       East Asia    Asia   1  \n",
       "2         China_AmurRiver_EarlyN       East Asia    Asia   1  \n",
       "3  Albania_EarlyModern_oCaucasus  Eastern Europe  Europe   2  \n",
       "4             Armenia_Aknashen_N    Central Asia    Asia  -1  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genotypes_header = ['rsID','sample','gt']\n",
    "rs1790218_proxy_variant_genotypes = pd.read_csv('/wynton/group/capra/projects/SLC22A10/data/rs1790218_proxy_variants.geno', delim_whitespace = True, names = genotypes_header)\n",
    "rs1790218_proxy_variant_genotypes = rs1790218_proxy_variant_genotypes[rs1790218_proxy_variant_genotypes['rsID'] != 'Affx-5633112']\n",
    "rs1790218_proxy_variant_genotypes['sample_ID'] = rs1790218_proxy_variant_genotypes['sample'].map(IDs_dict)\n",
    "rs1790218_proxy_variant_genotypes['date'] = rs1790218_proxy_variant_genotypes['sample'].map(dates_dict)\n",
    "rs1790218_proxy_variant_genotypes['location'] = rs1790218_proxy_variant_genotypes['sample'].map(location_dict)\n",
    "rs1790218_proxy_variant_genotypes['subregion'] = rs1790218_proxy_variant_genotypes['sample'].map(subregion_dict)\n",
    "rs1790218_proxy_variant_genotypes['region'] = rs1790218_proxy_variant_genotypes['sample'].map(region_dict)\n",
    "rs1790218_proxy_variant_genotypes = rs1790218_proxy_variant_genotypes[['rsID','sample_ID','sample','date','location','subregion','region','gt']]\n",
    "rs1790218_proxy_variant_genotypes.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3af47000-9d45-4083-8868-bb946ec1b757",
   "metadata": {},
   "source": [
    "First, let's subset the archaic hominin samples to determine if any non-reference alleles are present."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "ab033af1-b6ce-440c-93a9-9077f241d673",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsID</th>\n",
       "      <th>sample_ID</th>\n",
       "      <th>sample</th>\n",
       "      <th>date</th>\n",
       "      <th>location</th>\n",
       "      <th>subregion</th>\n",
       "      <th>region</th>\n",
       "      <th>gt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>156</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>Altai</td>\n",
       "      <td>AltaiNeanderthal.DG</td>\n",
       "      <td>110450.0</td>\n",
       "      <td>Altai_Neanderthal.DG</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>Altai</td>\n",
       "      <td>AltaiNeanderthal_snpAD.DG</td>\n",
       "      <td>110450.0</td>\n",
       "      <td>Altai_Neanderthal.DG</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1059</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>Chagyrskaya</td>\n",
       "      <td>Chagyrskaya_noUDG.SG</td>\n",
       "      <td>80000.0</td>\n",
       "      <td>Chagyrskaya_Neanderthal.SG</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2001</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>Denisova</td>\n",
       "      <td>Denisova3.DG</td>\n",
       "      <td>63900.0</td>\n",
       "      <td>Denisova.DG</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2002</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>Denisova</td>\n",
       "      <td>Denisova3_snpAD.DG</td>\n",
       "      <td>63900.0</td>\n",
       "      <td>Denisova.DG</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2003</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>Denisova11</td>\n",
       "      <td>Denisova11_noUDG.SG</td>\n",
       "      <td>98700.0</td>\n",
       "      <td>DenisovaNeanderthalMix.SG</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3692</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>Goyet_final</td>\n",
       "      <td>Goyet_noUDG.SG</td>\n",
       "      <td>42436.0</td>\n",
       "      <td>Goyet_Neanderthal.SG</td>\n",
       "      <td>Western Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5769</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>Les_Cottes_final</td>\n",
       "      <td>Les_Cottes_noUDG.SG</td>\n",
       "      <td>42846.0</td>\n",
       "      <td>LesCottes_Neanderthal.SG</td>\n",
       "      <td>Western Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6022</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>Mezmaiskaya2_final</td>\n",
       "      <td>Mezmaiskaya2_noUDG.SG</td>\n",
       "      <td>43449.0</td>\n",
       "      <td>Mezmaiskaya2_Neanderthal.SG</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8385</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>Spy_final</td>\n",
       "      <td>Spy_noUDG.SG</td>\n",
       "      <td>40922.0</td>\n",
       "      <td>Spy_Neanderthal.SG</td>\n",
       "      <td>Western Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9431</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>Vindija</td>\n",
       "      <td>Vindija_snpAD.DG</td>\n",
       "      <td>50000.0</td>\n",
       "      <td>Vindija_Neanderthal.DG</td>\n",
       "      <td>Southern Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9432</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>VindijaG1_final</td>\n",
       "      <td>VindijaG1_noUDG.SG</td>\n",
       "      <td>50000.0</td>\n",
       "      <td>VindijaG1_Neanderthal.SG</td>\n",
       "      <td>Southern Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26190</th>\n",
       "      <td>rs1201559</td>\n",
       "      <td>Altai</td>\n",
       "      <td>AltaiNeanderthal.DG</td>\n",
       "      <td>110450.0</td>\n",
       "      <td>Altai_Neanderthal.DG</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26191</th>\n",
       "      <td>rs1201559</td>\n",
       "      <td>Altai</td>\n",
       "      <td>AltaiNeanderthal_snpAD.DG</td>\n",
       "      <td>110450.0</td>\n",
       "      <td>Altai_Neanderthal.DG</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27093</th>\n",
       "      <td>rs1201559</td>\n",
       "      <td>Chagyrskaya</td>\n",
       "      <td>Chagyrskaya_noUDG.SG</td>\n",
       "      <td>80000.0</td>\n",
       "      <td>Chagyrskaya_Neanderthal.SG</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28035</th>\n",
       "      <td>rs1201559</td>\n",
       "      <td>Denisova</td>\n",
       "      <td>Denisova3.DG</td>\n",
       "      <td>63900.0</td>\n",
       "      <td>Denisova.DG</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28036</th>\n",
       "      <td>rs1201559</td>\n",
       "      <td>Denisova</td>\n",
       "      <td>Denisova3_snpAD.DG</td>\n",
       "      <td>63900.0</td>\n",
       "      <td>Denisova.DG</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28037</th>\n",
       "      <td>rs1201559</td>\n",
       "      <td>Denisova11</td>\n",
       "      <td>Denisova11_noUDG.SG</td>\n",
       "      <td>98700.0</td>\n",
       "      <td>DenisovaNeanderthalMix.SG</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29726</th>\n",
       "      <td>rs1201559</td>\n",
       "      <td>Goyet_final</td>\n",
       "      <td>Goyet_noUDG.SG</td>\n",
       "      <td>42436.0</td>\n",
       "      <td>Goyet_Neanderthal.SG</td>\n",
       "      <td>Western Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31803</th>\n",
       "      <td>rs1201559</td>\n",
       "      <td>Les_Cottes_final</td>\n",
       "      <td>Les_Cottes_noUDG.SG</td>\n",
       "      <td>42846.0</td>\n",
       "      <td>LesCottes_Neanderthal.SG</td>\n",
       "      <td>Western Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32056</th>\n",
       "      <td>rs1201559</td>\n",
       "      <td>Mezmaiskaya2_final</td>\n",
       "      <td>Mezmaiskaya2_noUDG.SG</td>\n",
       "      <td>43449.0</td>\n",
       "      <td>Mezmaiskaya2_Neanderthal.SG</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34419</th>\n",
       "      <td>rs1201559</td>\n",
       "      <td>Spy_final</td>\n",
       "      <td>Spy_noUDG.SG</td>\n",
       "      <td>40922.0</td>\n",
       "      <td>Spy_Neanderthal.SG</td>\n",
       "      <td>Western Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35465</th>\n",
       "      <td>rs1201559</td>\n",
       "      <td>Vindija</td>\n",
       "      <td>Vindija_snpAD.DG</td>\n",
       "      <td>50000.0</td>\n",
       "      <td>Vindija_Neanderthal.DG</td>\n",
       "      <td>Southern Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35466</th>\n",
       "      <td>rs1201559</td>\n",
       "      <td>VindijaG1_final</td>\n",
       "      <td>VindijaG1_noUDG.SG</td>\n",
       "      <td>50000.0</td>\n",
       "      <td>VindijaG1_Neanderthal.SG</td>\n",
       "      <td>Southern Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            rsID           sample_ID                     sample      date  \\\n",
       "156    rs1783634               Altai        AltaiNeanderthal.DG  110450.0   \n",
       "157    rs1783634               Altai  AltaiNeanderthal_snpAD.DG  110450.0   \n",
       "1059   rs1783634         Chagyrskaya       Chagyrskaya_noUDG.SG   80000.0   \n",
       "2001   rs1783634            Denisova               Denisova3.DG   63900.0   \n",
       "2002   rs1783634            Denisova         Denisova3_snpAD.DG   63900.0   \n",
       "2003   rs1783634          Denisova11        Denisova11_noUDG.SG   98700.0   \n",
       "3692   rs1783634         Goyet_final             Goyet_noUDG.SG   42436.0   \n",
       "5769   rs1783634    Les_Cottes_final        Les_Cottes_noUDG.SG   42846.0   \n",
       "6022   rs1783634  Mezmaiskaya2_final      Mezmaiskaya2_noUDG.SG   43449.0   \n",
       "8385   rs1783634           Spy_final               Spy_noUDG.SG   40922.0   \n",
       "9431   rs1783634             Vindija           Vindija_snpAD.DG   50000.0   \n",
       "9432   rs1783634     VindijaG1_final         VindijaG1_noUDG.SG   50000.0   \n",
       "26190  rs1201559               Altai        AltaiNeanderthal.DG  110450.0   \n",
       "26191  rs1201559               Altai  AltaiNeanderthal_snpAD.DG  110450.0   \n",
       "27093  rs1201559         Chagyrskaya       Chagyrskaya_noUDG.SG   80000.0   \n",
       "28035  rs1201559            Denisova               Denisova3.DG   63900.0   \n",
       "28036  rs1201559            Denisova         Denisova3_snpAD.DG   63900.0   \n",
       "28037  rs1201559          Denisova11        Denisova11_noUDG.SG   98700.0   \n",
       "29726  rs1201559         Goyet_final             Goyet_noUDG.SG   42436.0   \n",
       "31803  rs1201559    Les_Cottes_final        Les_Cottes_noUDG.SG   42846.0   \n",
       "32056  rs1201559  Mezmaiskaya2_final      Mezmaiskaya2_noUDG.SG   43449.0   \n",
       "34419  rs1201559           Spy_final               Spy_noUDG.SG   40922.0   \n",
       "35465  rs1201559             Vindija           Vindija_snpAD.DG   50000.0   \n",
       "35466  rs1201559     VindijaG1_final         VindijaG1_noUDG.SG   50000.0   \n",
       "\n",
       "                          location        subregion  region  gt  \n",
       "156           Altai_Neanderthal.DG    Northern Asia    Asia   2  \n",
       "157           Altai_Neanderthal.DG    Northern Asia    Asia   2  \n",
       "1059    Chagyrskaya_Neanderthal.SG    Northern Asia    Asia   2  \n",
       "2001                   Denisova.DG    Northern Asia    Asia   2  \n",
       "2002                   Denisova.DG    Northern Asia    Asia   2  \n",
       "2003     DenisovaNeanderthalMix.SG    Northern Asia    Asia  -1  \n",
       "3692          Goyet_Neanderthal.SG   Western Europe  Europe   2  \n",
       "5769      LesCottes_Neanderthal.SG   Western Europe  Europe   2  \n",
       "6022   Mezmaiskaya2_Neanderthal.SG    Northern Asia    Asia   2  \n",
       "8385            Spy_Neanderthal.SG   Western Europe  Europe  -1  \n",
       "9431        Vindija_Neanderthal.DG  Southern Europe  Europe   2  \n",
       "9432      VindijaG1_Neanderthal.SG  Southern Europe  Europe   2  \n",
       "26190         Altai_Neanderthal.DG    Northern Asia    Asia   2  \n",
       "26191         Altai_Neanderthal.DG    Northern Asia    Asia   2  \n",
       "27093   Chagyrskaya_Neanderthal.SG    Northern Asia    Asia   2  \n",
       "28035                  Denisova.DG    Northern Asia    Asia   2  \n",
       "28036                  Denisova.DG    Northern Asia    Asia   2  \n",
       "28037    DenisovaNeanderthalMix.SG    Northern Asia    Asia   2  \n",
       "29726         Goyet_Neanderthal.SG   Western Europe  Europe   2  \n",
       "31803     LesCottes_Neanderthal.SG   Western Europe  Europe   2  \n",
       "32056  Mezmaiskaya2_Neanderthal.SG    Northern Asia    Asia   2  \n",
       "34419           Spy_Neanderthal.SG   Western Europe  Europe   2  \n",
       "35465       Vindija_Neanderthal.DG  Southern Europe  Europe   2  \n",
       "35466     VindijaG1_Neanderthal.SG  Southern Europe  Europe   2  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rs1790218_proxy_variant_genotypes[(rs1790218_proxy_variant_genotypes['location'].str.contains('Denisova')) | (rs1790218_proxy_variant_genotypes['location'].str.contains('Neanderthal'))]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0f3b142c-39ef-41c5-b44a-9d5f06235048",
   "metadata": {},
   "source": [
    "It appears that the archaic hominins are fixed for the reference allele at both proxy SNVs. Let's remove these from the dataframe so that we can focus on anatomically modern *Homo sapiens*. Check that this filtering was successful."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "348d6d24-490a-41f7-b43d-ed4cf2052761",
   "metadata": {},
   "outputs": [],
   "source": [
    "rs1790218_proxy_variant_genotypes = rs1790218_proxy_variant_genotypes.drop(rs1790218_proxy_variant_genotypes[(rs1790218_proxy_variant_genotypes['location'].str.contains('Denisova')) | (rs1790218_proxy_variant_genotypes['location'].str.contains('Neanderthal'))].index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "3e5d0e11-c5c6-405b-b7d8-e10bb955979c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsID</th>\n",
       "      <th>sample_ID</th>\n",
       "      <th>sample</th>\n",
       "      <th>date</th>\n",
       "      <th>location</th>\n",
       "      <th>subregion</th>\n",
       "      <th>region</th>\n",
       "      <th>gt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [rsID, sample_ID, sample, date, location, subregion, region, gt]\n",
       "Index: []"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rs1790218_proxy_variant_genotypes[(rs1790218_proxy_variant_genotypes['location'].str.contains('Denisova')) | (rs1790218_proxy_variant_genotypes['location'].str.contains('Neanderthal'))]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "acaab2f4-0b5c-4313-bf47-e8fdd7308fd4",
   "metadata": {},
   "source": [
    "Now look at the genotype distribution."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "9d3dd650-9331-43f6-b5f7-6a1b635d1b61",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rsID</th>\n",
       "      <th>gt</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">rs1201559</th>\n",
       "      <th>-1</th>\n",
       "      <td>1061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4769</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1724</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5451</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">rs1783634</th>\n",
       "      <th>-1</th>\n",
       "      <td>4735</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1724</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3545</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 N\n",
       "rsID      gt      \n",
       "rs1201559 -1  1061\n",
       "           0  4769\n",
       "           1  1724\n",
       "           2  5451\n",
       "rs1783634 -1  4735\n",
       "           0  3001\n",
       "           1  1724\n",
       "           2  3545"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rs1790218_proxy_variant_genotypes.groupby(['rsID','gt']).size().to_frame('N')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f35370e9-e4e7-44ba-a2d9-fefb827abf47",
   "metadata": {},
   "source": [
    "Given the presence of missing genotypes (gt == -1), let's filter those out and retain only individuals with genotypes at both loci."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "5131a50f-e66c-4bcc-8b65-2396b51e48e9",
   "metadata": {},
   "outputs": [],
   "source": [
    "nonmissing_rs1790218_proxy_variant_genotypes = rs1790218_proxy_variant_genotypes[rs1790218_proxy_variant_genotypes['gt'] != -1]\n",
    "full_coverage_rs1790218_proxy_variant_genotypes = nonmissing_rs1790218_proxy_variant_genotypes.groupby('sample').filter(lambda x: len(x) > 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "35c29f61-8d11-46e4-b4df-d4293b2f17e9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "7744.0"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(full_coverage_rs1790218_proxy_variant_genotypes)/2"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8260500e-6928-48c4-bce8-6835436e4855",
   "metadata": {},
   "source": [
    "Let's assess how many duplicate entries there are per sample."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "09fc42bd-d06c-4e9f-ac7e-3b62ea76c10d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2659.0"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_coverage_rs1790218_proxy_variant_genotypes.duplicated(subset=['rsID','sample_ID']).sum()/2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "6b7b18b2-db61-461d-afb1-c2d89e77b1a8",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsID</th>\n",
       "      <th>sample_ID</th>\n",
       "      <th>sample</th>\n",
       "      <th>date</th>\n",
       "      <th>location</th>\n",
       "      <th>subregion</th>\n",
       "      <th>region</th>\n",
       "      <th>gt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>132</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HGDP01401</td>\n",
       "      <td>HGDP01401.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Adygei.DG</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HGDP01402</td>\n",
       "      <td>HGDP01402.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Adygei.DG</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>461</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HGDP00058</td>\n",
       "      <td>HGDP00058.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Balochi.DG</td>\n",
       "      <td>South Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>472</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HGDP00090</td>\n",
       "      <td>HGDP00090.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Balochi.DG</td>\n",
       "      <td>South Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>484</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HGDP01414</td>\n",
       "      <td>HGDP01414.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>BantuKenya.DG</td>\n",
       "      <td>Eastern Africa</td>\n",
       "      <td>Africa</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>485</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HGDP01417</td>\n",
       "      <td>HGDP01417.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>BantuKenya.DG</td>\n",
       "      <td>Eastern Africa</td>\n",
       "      <td>Africa</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>488</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HGDP01028</td>\n",
       "      <td>HGDP01028.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>BantuSA_Herero.DG</td>\n",
       "      <td>Southern Africa</td>\n",
       "      <td>Africa</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>489</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HGDP01035</td>\n",
       "      <td>HGDP01035.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>BantuSA_Herero.DG</td>\n",
       "      <td>Southern Africa</td>\n",
       "      <td>Africa</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>496</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HGDP01030</td>\n",
       "      <td>S_BantuTswana-1.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>BantuTswana.DG</td>\n",
       "      <td>Southern Africa</td>\n",
       "      <td>Africa</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>497</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HGDP01034</td>\n",
       "      <td>S_BantuTswana-2.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>BantuTswana.DG</td>\n",
       "      <td>Southern Africa</td>\n",
       "      <td>Africa</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          rsID  sample_ID              sample  date           location  \\\n",
       "132  rs1783634  HGDP01401        HGDP01401.DG   0.0          Adygei.DG   \n",
       "133  rs1783634  HGDP01402        HGDP01402.DG   0.0          Adygei.DG   \n",
       "461  rs1783634  HGDP00058        HGDP00058.DG   0.0         Balochi.DG   \n",
       "472  rs1783634  HGDP00090        HGDP00090.DG   0.0         Balochi.DG   \n",
       "484  rs1783634  HGDP01414        HGDP01414.DG   0.0      BantuKenya.DG   \n",
       "485  rs1783634  HGDP01417        HGDP01417.DG   0.0      BantuKenya.DG   \n",
       "488  rs1783634  HGDP01028        HGDP01028.DG   0.0  BantuSA_Herero.DG   \n",
       "489  rs1783634  HGDP01035        HGDP01035.DG   0.0  BantuSA_Herero.DG   \n",
       "496  rs1783634  HGDP01030  S_BantuTswana-1.DG   0.0     BantuTswana.DG   \n",
       "497  rs1783634  HGDP01034  S_BantuTswana-2.DG   0.0     BantuTswana.DG   \n",
       "\n",
       "           subregion  region  gt  \n",
       "132    Northern Asia    Asia   1  \n",
       "133    Northern Asia    Asia   1  \n",
       "461       South Asia    Asia   0  \n",
       "472       South Asia    Asia   1  \n",
       "484   Eastern Africa  Africa   2  \n",
       "485   Eastern Africa  Africa   1  \n",
       "488  Southern Africa  Africa   2  \n",
       "489  Southern Africa  Africa   0  \n",
       "496  Southern Africa  Africa   2  \n",
       "497  Southern Africa  Africa   1  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_coverage_rs1790218_proxy_variant_genotypes[full_coverage_rs1790218_proxy_variant_genotypes.duplicated(subset=['rsID','sample_ID'])].head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a9ccfae2-c848-4554-b39d-c815e8795e29",
   "metadata": {},
   "source": [
    "Many of these appear to be samples from Thousand Genomes (1KG) and the Simons Genome Diversity Project (SGDP). Let's keep the 1KG samples with read calls that allow for heterozygous genotypes (DG) and the newer SGDP genotypes from Bergstrom et al. 2020 whose sample_ID starts with HGDP."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "c6b5b854-a8dd-4603-bf3b-402541ab2022",
   "metadata": {},
   "outputs": [],
   "source": [
    "full_coverage_rs1790218_proxy_variant_genotypes = full_coverage_rs1790218_proxy_variant_genotypes.drop(full_coverage_rs1790218_proxy_variant_genotypes[(full_coverage_rs1790218_proxy_variant_genotypes['sample'].str.startswith(('HG0','NA'))) & (full_coverage_rs1790218_proxy_variant_genotypes['sample'].str.endswith('SG'))].index)\n",
    "full_coverage_rs1790218_proxy_variant_genotypes = full_coverage_rs1790218_proxy_variant_genotypes.drop(full_coverage_rs1790218_proxy_variant_genotypes[(full_coverage_rs1790218_proxy_variant_genotypes['sample_ID'].str.startswith('HGDP')) & (~full_coverage_rs1790218_proxy_variant_genotypes['sample'].str.startswith('HGDP'))].index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "efb0424a-e49e-40a9-bdf0-9bba5c8f0e03",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5102.0"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(full_coverage_rs1790218_proxy_variant_genotypes)/2"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "28a1832b-c069-4dbf-a3ee-ca3f49464c1b",
   "metadata": {},
   "source": [
    "How many duplicates does that leave?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "6a133654-21c4-4be5-b87a-2889ae34b05a",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "49.0"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_coverage_rs1790218_proxy_variant_genotypes.duplicated(subset=['rsID','sample_ID']).sum()/2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "76505df0-aa3a-40f4-a79f-621160f84b8e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsID</th>\n",
       "      <th>sample_ID</th>\n",
       "      <th>sample</th>\n",
       "      <th>date</th>\n",
       "      <th>location</th>\n",
       "      <th>subregion</th>\n",
       "      <th>region</th>\n",
       "      <th>gt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>848</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I10871</td>\n",
       "      <td>I10871_noUDG</td>\n",
       "      <td>7913.0</td>\n",
       "      <td>Cameroon_SMA_noUDG</td>\n",
       "      <td>Central Africa</td>\n",
       "      <td>Africa</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>850</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I10873</td>\n",
       "      <td>I10873_noUDG</td>\n",
       "      <td>3097.0</td>\n",
       "      <td>Cameroon_SMA_noUDG</td>\n",
       "      <td>Central Africa</td>\n",
       "      <td>Africa</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>851</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I10874</td>\n",
       "      <td>I10874_noUDG</td>\n",
       "      <td>3135.0</td>\n",
       "      <td>Cameroon_SMA_noUDG</td>\n",
       "      <td>Central Africa</td>\n",
       "      <td>Africa</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>853</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I10871</td>\n",
       "      <td>I10871_noUDG.DG</td>\n",
       "      <td>7913.0</td>\n",
       "      <td>Cameroon_SMA.DG</td>\n",
       "      <td>Central Africa</td>\n",
       "      <td>Africa</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>854</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I10871</td>\n",
       "      <td>I10871_noUDG.SG</td>\n",
       "      <td>7913.0</td>\n",
       "      <td>Cameroon_SMA.SG</td>\n",
       "      <td>Central Africa</td>\n",
       "      <td>Africa</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>855</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I10873</td>\n",
       "      <td>I10873.SG</td>\n",
       "      <td>3097.0</td>\n",
       "      <td>Cameroon_SMA.SG</td>\n",
       "      <td>Central Africa</td>\n",
       "      <td>Africa</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2530</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I6767</td>\n",
       "      <td>I6767.SG</td>\n",
       "      <td>10280.0</td>\n",
       "      <td>England_Mesolithic.SG</td>\n",
       "      <td>Western Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2708</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I6746</td>\n",
       "      <td>I6746.SG</td>\n",
       "      <td>5526.0</td>\n",
       "      <td>England_N.SG</td>\n",
       "      <td>Western Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2989</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I5950</td>\n",
       "      <td>I5950.DG</td>\n",
       "      <td>4470.0</td>\n",
       "      <td>Ethiopia_4500BP.DG</td>\n",
       "      <td>Northern Africa</td>\n",
       "      <td>Africa</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2990</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I5950</td>\n",
       "      <td>I5950.SG</td>\n",
       "      <td>4470.0</td>\n",
       "      <td>Ethiopia_4500BP.SG</td>\n",
       "      <td>Northern Africa</td>\n",
       "      <td>Africa</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4515</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I2323</td>\n",
       "      <td>I2323_noUDG</td>\n",
       "      <td>7910.0</td>\n",
       "      <td>Iran_HajjiFiruz_N_noUDG</td>\n",
       "      <td>Western Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5790</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I0001</td>\n",
       "      <td>Loschbour_snpAD.DG</td>\n",
       "      <td>8025.0</td>\n",
       "      <td>Luxembourg_Loschbour.DG</td>\n",
       "      <td>Western Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5994</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HGDP00877</td>\n",
       "      <td>HGDP00877.SG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Mayan.SG</td>\n",
       "      <td>Americas</td>\n",
       "      <td>Americas</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7423</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I6714</td>\n",
       "      <td>I6714.DG</td>\n",
       "      <td>4479.0</td>\n",
       "      <td>Russia_Afanasievo_son.I3388_son.I3950_brother....</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7424</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I6714</td>\n",
       "      <td>I6714.SG</td>\n",
       "      <td>4479.0</td>\n",
       "      <td>Russia_Afanasievo_son.I3388_son.I3950_brother....</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7427</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I3949</td>\n",
       "      <td>I3949.SG</td>\n",
       "      <td>4569.0</td>\n",
       "      <td>Russia_Afanasievo_son.I3388_son.I3950_brother....</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7428</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I3388</td>\n",
       "      <td>I3388.DG</td>\n",
       "      <td>4650.0</td>\n",
       "      <td>Russia_Afanasievo.DG</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7431</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I3388</td>\n",
       "      <td>I3388.SG</td>\n",
       "      <td>4650.0</td>\n",
       "      <td>Russia_Afanasievo.SG</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7432</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I3950</td>\n",
       "      <td>I3950.SG</td>\n",
       "      <td>4707.0</td>\n",
       "      <td>Russia_Afanasievo.SG</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7782</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>Ust_Ishim</td>\n",
       "      <td>UstIshim_snpAD.DG</td>\n",
       "      <td>44366.0</td>\n",
       "      <td>Russia_Ust_Ishim.DG</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8262</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I0585</td>\n",
       "      <td>LaBrana1_noUDG.SG</td>\n",
       "      <td>7811.0</td>\n",
       "      <td>Spain_HG.SG</td>\n",
       "      <td>Western Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8555</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>vbj001</td>\n",
       "      <td>vbj001.SG</td>\n",
       "      <td>4675.0</td>\n",
       "      <td>Sweden_Gotland_Vasterbjers_PittedWare_BattleAx...</td>\n",
       "      <td>Northern Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8556</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>vbj004</td>\n",
       "      <td>vbj004.SG</td>\n",
       "      <td>4702.0</td>\n",
       "      <td>Sweden_Gotland_Vasterbjers_PittedWare_BattleAx...</td>\n",
       "      <td>Northern Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8559</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>vbj012</td>\n",
       "      <td>vbj012.SG</td>\n",
       "      <td>4933.0</td>\n",
       "      <td>Sweden_Gotland_Vasterbjers_PittedWare_BattleAx...</td>\n",
       "      <td>Northern Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9091</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I1583</td>\n",
       "      <td>I1583</td>\n",
       "      <td>8273.0</td>\n",
       "      <td>Turkey_N</td>\n",
       "      <td>Western Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9093</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I1583</td>\n",
       "      <td>I1583.DG</td>\n",
       "      <td>8273.0</td>\n",
       "      <td>Turkey_N.DG</td>\n",
       "      <td>Western Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9096</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I1583</td>\n",
       "      <td>I1583.SG</td>\n",
       "      <td>8273.0</td>\n",
       "      <td>Turkey_N.SG</td>\n",
       "      <td>Western Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9251</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I5319</td>\n",
       "      <td>I5319.SG</td>\n",
       "      <td>725.0</td>\n",
       "      <td>USA_AK_Ancient_Athabaskan_1100BP.SG</td>\n",
       "      <td>Americas</td>\n",
       "      <td>Americas</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10157</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HG02783</td>\n",
       "      <td>HG02783.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PJL.DG</td>\n",
       "      <td>South Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10318</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HG02464</td>\n",
       "      <td>HG02464.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GWD.DG</td>\n",
       "      <td>Western Africa</td>\n",
       "      <td>Africa</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10328</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HG01600</td>\n",
       "      <td>HG01600.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>KHV.DG</td>\n",
       "      <td>South Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10354</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HG02574</td>\n",
       "      <td>HG02574.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GWD.DG</td>\n",
       "      <td>Western Africa</td>\n",
       "      <td>Africa</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10511</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HG00128</td>\n",
       "      <td>HG00128.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GBR.DG</td>\n",
       "      <td>Western Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10546</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HG03007</td>\n",
       "      <td>HG03007.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>BEB.DG</td>\n",
       "      <td>South Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10855</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HG00174</td>\n",
       "      <td>HG00174.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>FIN.DG</td>\n",
       "      <td>Northern Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11101</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HG02790</td>\n",
       "      <td>HG02790.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PJL.DG</td>\n",
       "      <td>South Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11188</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HG00126</td>\n",
       "      <td>HG00126.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GBR.DG</td>\n",
       "      <td>Western Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11399</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HG01846</td>\n",
       "      <td>HG01846.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>KHV.DG</td>\n",
       "      <td>South Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11438</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HG03085</td>\n",
       "      <td>HG03085.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>MSL.DG</td>\n",
       "      <td>Western Africa</td>\n",
       "      <td>Africa</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11820</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HG02724</td>\n",
       "      <td>HG02724.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PJL.DG</td>\n",
       "      <td>South Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11883</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>NA19023</td>\n",
       "      <td>NA19023.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>LWK.DG</td>\n",
       "      <td>Eastern Africa</td>\n",
       "      <td>Africa</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11909</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HG00190</td>\n",
       "      <td>HG00190.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>FIN.DG</td>\n",
       "      <td>Northern Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11930</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HG03100</td>\n",
       "      <td>HG03100.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ESN.DG</td>\n",
       "      <td>Northern Africa</td>\n",
       "      <td>Africa</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12089</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HG01503</td>\n",
       "      <td>HG01503.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IBS.DG</td>\n",
       "      <td>Western Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12090</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HG01504</td>\n",
       "      <td>HG01504.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IBS.DG</td>\n",
       "      <td>Western Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12239</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HG02943</td>\n",
       "      <td>HG02943.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ESN.DG</td>\n",
       "      <td>Northern Africa</td>\n",
       "      <td>Africa</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12250</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HG03006</td>\n",
       "      <td>HG03006.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>BEB.DG</td>\n",
       "      <td>South Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12351</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HG03078</td>\n",
       "      <td>HG03078.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>MSL.DG</td>\n",
       "      <td>Western Africa</td>\n",
       "      <td>Africa</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13010</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I15620</td>\n",
       "      <td>I15620_v54.1_addback</td>\n",
       "      <td>5950.0</td>\n",
       "      <td>Romania_C_Bodrogkeresztur</td>\n",
       "      <td>Eastern Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            rsID  sample_ID                sample     date  \\\n",
       "848    rs1783634     I10871          I10871_noUDG   7913.0   \n",
       "850    rs1783634     I10873          I10873_noUDG   3097.0   \n",
       "851    rs1783634     I10874          I10874_noUDG   3135.0   \n",
       "853    rs1783634     I10871       I10871_noUDG.DG   7913.0   \n",
       "854    rs1783634     I10871       I10871_noUDG.SG   7913.0   \n",
       "855    rs1783634     I10873             I10873.SG   3097.0   \n",
       "2530   rs1783634      I6767              I6767.SG  10280.0   \n",
       "2708   rs1783634      I6746              I6746.SG   5526.0   \n",
       "2989   rs1783634      I5950              I5950.DG   4470.0   \n",
       "2990   rs1783634      I5950              I5950.SG   4470.0   \n",
       "4515   rs1783634      I2323           I2323_noUDG   7910.0   \n",
       "5790   rs1783634      I0001    Loschbour_snpAD.DG   8025.0   \n",
       "5994   rs1783634  HGDP00877          HGDP00877.SG      0.0   \n",
       "7423   rs1783634      I6714              I6714.DG   4479.0   \n",
       "7424   rs1783634      I6714              I6714.SG   4479.0   \n",
       "7427   rs1783634      I3949              I3949.SG   4569.0   \n",
       "7428   rs1783634      I3388              I3388.DG   4650.0   \n",
       "7431   rs1783634      I3388              I3388.SG   4650.0   \n",
       "7432   rs1783634      I3950              I3950.SG   4707.0   \n",
       "7782   rs1783634  Ust_Ishim     UstIshim_snpAD.DG  44366.0   \n",
       "8262   rs1783634      I0585     LaBrana1_noUDG.SG   7811.0   \n",
       "8555   rs1783634     vbj001             vbj001.SG   4675.0   \n",
       "8556   rs1783634     vbj004             vbj004.SG   4702.0   \n",
       "8559   rs1783634     vbj012             vbj012.SG   4933.0   \n",
       "9091   rs1783634      I1583                 I1583   8273.0   \n",
       "9093   rs1783634      I1583              I1583.DG   8273.0   \n",
       "9096   rs1783634      I1583              I1583.SG   8273.0   \n",
       "9251   rs1783634      I5319              I5319.SG    725.0   \n",
       "10157  rs1783634    HG02783            HG02783.DG      0.0   \n",
       "10318  rs1783634    HG02464            HG02464.DG      0.0   \n",
       "10328  rs1783634    HG01600            HG01600.DG      0.0   \n",
       "10354  rs1783634    HG02574            HG02574.DG      0.0   \n",
       "10511  rs1783634    HG00128            HG00128.DG      0.0   \n",
       "10546  rs1783634    HG03007            HG03007.DG      0.0   \n",
       "10855  rs1783634    HG00174            HG00174.DG      0.0   \n",
       "11101  rs1783634    HG02790            HG02790.DG      0.0   \n",
       "11188  rs1783634    HG00126            HG00126.DG      0.0   \n",
       "11399  rs1783634    HG01846            HG01846.DG      0.0   \n",
       "11438  rs1783634    HG03085            HG03085.DG      0.0   \n",
       "11820  rs1783634    HG02724            HG02724.DG      0.0   \n",
       "11883  rs1783634    NA19023            NA19023.DG      0.0   \n",
       "11909  rs1783634    HG00190            HG00190.DG      0.0   \n",
       "11930  rs1783634    HG03100            HG03100.DG      0.0   \n",
       "12089  rs1783634    HG01503            HG01503.DG      0.0   \n",
       "12090  rs1783634    HG01504            HG01504.DG      0.0   \n",
       "12239  rs1783634    HG02943            HG02943.DG      0.0   \n",
       "12250  rs1783634    HG03006            HG03006.DG      0.0   \n",
       "12351  rs1783634    HG03078            HG03078.DG      0.0   \n",
       "13010  rs1783634     I15620  I15620_v54.1_addback   5950.0   \n",
       "\n",
       "                                                location        subregion  \\\n",
       "848                                   Cameroon_SMA_noUDG   Central Africa   \n",
       "850                                   Cameroon_SMA_noUDG   Central Africa   \n",
       "851                                   Cameroon_SMA_noUDG   Central Africa   \n",
       "853                                      Cameroon_SMA.DG   Central Africa   \n",
       "854                                      Cameroon_SMA.SG   Central Africa   \n",
       "855                                      Cameroon_SMA.SG   Central Africa   \n",
       "2530                               England_Mesolithic.SG   Western Europe   \n",
       "2708                                        England_N.SG   Western Europe   \n",
       "2989                                  Ethiopia_4500BP.DG  Northern Africa   \n",
       "2990                                  Ethiopia_4500BP.SG  Northern Africa   \n",
       "4515                             Iran_HajjiFiruz_N_noUDG     Western Asia   \n",
       "5790                             Luxembourg_Loschbour.DG   Western Europe   \n",
       "5994                                            Mayan.SG         Americas   \n",
       "7423   Russia_Afanasievo_son.I3388_son.I3950_brother....    Northern Asia   \n",
       "7424   Russia_Afanasievo_son.I3388_son.I3950_brother....    Northern Asia   \n",
       "7427   Russia_Afanasievo_son.I3388_son.I3950_brother....    Northern Asia   \n",
       "7428                                Russia_Afanasievo.DG    Northern Asia   \n",
       "7431                                Russia_Afanasievo.SG    Northern Asia   \n",
       "7432                                Russia_Afanasievo.SG    Northern Asia   \n",
       "7782                                 Russia_Ust_Ishim.DG    Northern Asia   \n",
       "8262                                         Spain_HG.SG   Western Europe   \n",
       "8555   Sweden_Gotland_Vasterbjers_PittedWare_BattleAx...  Northern Europe   \n",
       "8556   Sweden_Gotland_Vasterbjers_PittedWare_BattleAx...  Northern Europe   \n",
       "8559   Sweden_Gotland_Vasterbjers_PittedWare_BattleAx...  Northern Europe   \n",
       "9091                                            Turkey_N     Western Asia   \n",
       "9093                                         Turkey_N.DG     Western Asia   \n",
       "9096                                         Turkey_N.SG     Western Asia   \n",
       "9251                 USA_AK_Ancient_Athabaskan_1100BP.SG         Americas   \n",
       "10157                                             PJL.DG       South Asia   \n",
       "10318                                             GWD.DG   Western Africa   \n",
       "10328                                             KHV.DG       South Asia   \n",
       "10354                                             GWD.DG   Western Africa   \n",
       "10511                                             GBR.DG   Western Europe   \n",
       "10546                                             BEB.DG       South Asia   \n",
       "10855                                             FIN.DG  Northern Europe   \n",
       "11101                                             PJL.DG       South Asia   \n",
       "11188                                             GBR.DG   Western Europe   \n",
       "11399                                             KHV.DG       South Asia   \n",
       "11438                                             MSL.DG   Western Africa   \n",
       "11820                                             PJL.DG       South Asia   \n",
       "11883                                             LWK.DG   Eastern Africa   \n",
       "11909                                             FIN.DG  Northern Europe   \n",
       "11930                                             ESN.DG  Northern Africa   \n",
       "12089                                             IBS.DG   Western Europe   \n",
       "12090                                             IBS.DG   Western Europe   \n",
       "12239                                             ESN.DG  Northern Africa   \n",
       "12250                                             BEB.DG       South Asia   \n",
       "12351                                             MSL.DG   Western Africa   \n",
       "13010                          Romania_C_Bodrogkeresztur   Eastern Europe   \n",
       "\n",
       "         region  gt  \n",
       "848      Africa   0  \n",
       "850      Africa   0  \n",
       "851      Africa   2  \n",
       "853      Africa   0  \n",
       "854      Africa   0  \n",
       "855      Africa   0  \n",
       "2530     Europe   2  \n",
       "2708     Europe   0  \n",
       "2989     Africa   2  \n",
       "2990     Africa   2  \n",
       "4515       Asia   0  \n",
       "5790     Europe   2  \n",
       "5994   Americas   0  \n",
       "7423       Asia   2  \n",
       "7424       Asia   0  \n",
       "7427       Asia   0  \n",
       "7428       Asia   2  \n",
       "7431       Asia   2  \n",
       "7432       Asia   0  \n",
       "7782       Asia   2  \n",
       "8262     Europe   0  \n",
       "8555     Europe   0  \n",
       "8556     Europe   2  \n",
       "8559     Europe   0  \n",
       "9091       Asia   0  \n",
       "9093       Asia   0  \n",
       "9096       Asia   2  \n",
       "9251   Americas   2  \n",
       "10157      Asia   0  \n",
       "10318    Africa   1  \n",
       "10328      Asia   1  \n",
       "10354    Africa   2  \n",
       "10511    Europe   1  \n",
       "10546      Asia   1  \n",
       "10855    Europe   0  \n",
       "11101      Asia   2  \n",
       "11188    Europe   1  \n",
       "11399      Asia   0  \n",
       "11438    Africa   1  \n",
       "11820      Asia   2  \n",
       "11883    Africa   1  \n",
       "11909    Europe   1  \n",
       "11930    Africa   1  \n",
       "12089    Europe   1  \n",
       "12090    Europe   1  \n",
       "12239    Africa   2  \n",
       "12250      Asia   1  \n",
       "12351    Africa   2  \n",
       "13010    Europe   0  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_coverage_rs1790218_proxy_variant_genotypes[full_coverage_rs1790218_proxy_variant_genotypes.duplicated(subset=['rsID','sample_ID'])].head(49)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1d19de22-92ea-4ecb-87f2-aea3bf5d4728",
   "metadata": {},
   "source": [
    "First, let's keep the snpAD genotypes for Loschbour and Ust-Ishim."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "63bf0752-4896-4217-9cc4-69080e1e9367",
   "metadata": {},
   "outputs": [],
   "source": [
    "full_coverage_rs1790218_proxy_variant_genotypes = full_coverage_rs1790218_proxy_variant_genotypes[~full_coverage_rs1790218_proxy_variant_genotypes['sample'].isin(['Loschbour.DG','Ust_Ishim.DG'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "7679962b-5045-4d3e-9353-56a874e1a9f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10200"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(full_coverage_rs1790218_proxy_variant_genotypes)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "33f5d847-a831-468f-a035-6139bcc82e28",
   "metadata": {},
   "source": [
    "Now let's check if the remaining duplicates disagree in genotype. We can just keep whichever comes first for those that match and drop those that disagree."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "677e3205-f710-4ca6-bbc0-77456fc00f63",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "83"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_coverage_rs1790218_proxy_variant_genotypes.duplicated(subset=['rsID','sample_ID','gt']).sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "8bbd95cd-ed1f-48b1-bafa-99124113d2ec",
   "metadata": {},
   "outputs": [],
   "source": [
    "full_coverage_rs1790218_proxy_variant_genotypes = full_coverage_rs1790218_proxy_variant_genotypes.drop_duplicates(subset=['rsID','sample_ID','gt'], keep = 'first')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "1d1608c6-8e1d-4ca0-a66c-a1ff1ad6c274",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_coverage_rs1790218_proxy_variant_genotypes.duplicated(subset=['rsID','sample_ID']).sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "9275ffe4-0519-4781-99b7-42acc98e79a0",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsID</th>\n",
       "      <th>sample_ID</th>\n",
       "      <th>sample</th>\n",
       "      <th>date</th>\n",
       "      <th>location</th>\n",
       "      <th>subregion</th>\n",
       "      <th>region</th>\n",
       "      <th>gt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5994</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HGDP00877</td>\n",
       "      <td>HGDP00877.SG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Mayan.SG</td>\n",
       "      <td>Americas</td>\n",
       "      <td>Americas</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7424</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I6714</td>\n",
       "      <td>I6714.SG</td>\n",
       "      <td>4479.0</td>\n",
       "      <td>Russia_Afanasievo_son.I3388_son.I3950_brother....</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7427</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I3949</td>\n",
       "      <td>I3949.SG</td>\n",
       "      <td>4569.0</td>\n",
       "      <td>Russia_Afanasievo_son.I3388_son.I3950_brother....</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7432</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I3950</td>\n",
       "      <td>I3950.SG</td>\n",
       "      <td>4707.0</td>\n",
       "      <td>Russia_Afanasievo.SG</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8262</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I0585</td>\n",
       "      <td>LaBrana1_noUDG.SG</td>\n",
       "      <td>7811.0</td>\n",
       "      <td>Spain_HG.SG</td>\n",
       "      <td>Western Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9096</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I1583</td>\n",
       "      <td>I1583.SG</td>\n",
       "      <td>8273.0</td>\n",
       "      <td>Turkey_N.SG</td>\n",
       "      <td>Western Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9251</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>I5319</td>\n",
       "      <td>I5319.SG</td>\n",
       "      <td>725.0</td>\n",
       "      <td>USA_AK_Ancient_Athabaskan_1100BP.SG</td>\n",
       "      <td>Americas</td>\n",
       "      <td>Americas</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32028</th>\n",
       "      <td>rs1201559</td>\n",
       "      <td>HGDP00877</td>\n",
       "      <td>HGDP00877.SG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Mayan.SG</td>\n",
       "      <td>Americas</td>\n",
       "      <td>Americas</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33466</th>\n",
       "      <td>rs1201559</td>\n",
       "      <td>I3950</td>\n",
       "      <td>I3950.SG</td>\n",
       "      <td>4707.0</td>\n",
       "      <td>Russia_Afanasievo.SG</td>\n",
       "      <td>Northern Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34593</th>\n",
       "      <td>rs1201559</td>\n",
       "      <td>vbj012</td>\n",
       "      <td>vbj012.SG</td>\n",
       "      <td>4933.0</td>\n",
       "      <td>Sweden_Gotland_Vasterbjers_PittedWare_BattleAx...</td>\n",
       "      <td>Northern Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35125</th>\n",
       "      <td>rs1201559</td>\n",
       "      <td>I1583</td>\n",
       "      <td>I1583</td>\n",
       "      <td>8273.0</td>\n",
       "      <td>Turkey_N</td>\n",
       "      <td>Western Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            rsID  sample_ID             sample    date  \\\n",
       "5994   rs1783634  HGDP00877       HGDP00877.SG     0.0   \n",
       "7424   rs1783634      I6714           I6714.SG  4479.0   \n",
       "7427   rs1783634      I3949           I3949.SG  4569.0   \n",
       "7432   rs1783634      I3950           I3950.SG  4707.0   \n",
       "8262   rs1783634      I0585  LaBrana1_noUDG.SG  7811.0   \n",
       "9096   rs1783634      I1583           I1583.SG  8273.0   \n",
       "9251   rs1783634      I5319           I5319.SG   725.0   \n",
       "32028  rs1201559  HGDP00877       HGDP00877.SG     0.0   \n",
       "33466  rs1201559      I3950           I3950.SG  4707.0   \n",
       "34593  rs1201559     vbj012          vbj012.SG  4933.0   \n",
       "35125  rs1201559      I1583              I1583  8273.0   \n",
       "\n",
       "                                                location        subregion  \\\n",
       "5994                                            Mayan.SG         Americas   \n",
       "7424   Russia_Afanasievo_son.I3388_son.I3950_brother....    Northern Asia   \n",
       "7427   Russia_Afanasievo_son.I3388_son.I3950_brother....    Northern Asia   \n",
       "7432                                Russia_Afanasievo.SG    Northern Asia   \n",
       "8262                                         Spain_HG.SG   Western Europe   \n",
       "9096                                         Turkey_N.SG     Western Asia   \n",
       "9251                 USA_AK_Ancient_Athabaskan_1100BP.SG         Americas   \n",
       "32028                                           Mayan.SG         Americas   \n",
       "33466                               Russia_Afanasievo.SG    Northern Asia   \n",
       "34593  Sweden_Gotland_Vasterbjers_PittedWare_BattleAx...  Northern Europe   \n",
       "35125                                           Turkey_N     Western Asia   \n",
       "\n",
       "         region  gt  \n",
       "5994   Americas   0  \n",
       "7424       Asia   0  \n",
       "7427       Asia   0  \n",
       "7432       Asia   0  \n",
       "8262     Europe   0  \n",
       "9096       Asia   2  \n",
       "9251   Americas   2  \n",
       "32028  Americas   2  \n",
       "33466      Asia   0  \n",
       "34593    Europe   0  \n",
       "35125      Asia   2  "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_coverage_rs1790218_proxy_variant_genotypes[full_coverage_rs1790218_proxy_variant_genotypes.duplicated(subset=['rsID','sample_ID'])].head(11)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "3c4154a6-6a36-49ae-896b-4a060d73f417",
   "metadata": {},
   "outputs": [],
   "source": [
    "full_coverage_rs1790218_proxy_variant_genotypes = full_coverage_rs1790218_proxy_variant_genotypes[~full_coverage_rs1790218_proxy_variant_genotypes['sample_ID'].isin(['HGDP00877','I6714','I3949','I3950','I0585','I1583','I5319','HGDP00877','I3950','vbj012','I1583'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "1ee2d129-6498-4877-b3d6-3bb4942cc944",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_coverage_rs1790218_proxy_variant_genotypes.duplicated(subset=['rsID','sample_ID']).sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "182378c7-5b88-4819-8727-5f76223ecffc",
   "metadata": {},
   "source": [
    "How many samples are there after filtering? Ancient? Modern?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "422447ef-7648-45a9-8522-6d6c6031eea1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5045.0"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(full_coverage_rs1790218_proxy_variant_genotypes)/2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "3ac29c83-268a-4596-b4af-e671f81b297b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1375.0"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(full_coverage_rs1790218_proxy_variant_genotypes[full_coverage_rs1790218_proxy_variant_genotypes['date'] > 0])/2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "a1ea0da8-37c5-4fb8-8a93-5e32b2e72371",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3670.0"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(full_coverage_rs1790218_proxy_variant_genotypes[full_coverage_rs1790218_proxy_variant_genotypes['date'] == 0])/2"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "37d956a8-3910-4b75-8b88-05544f3253fc",
   "metadata": {},
   "source": [
    "Now summarize the genotypes and write a function to calculate allele frequency remembering that the AADR genotypes are 0 = homozygous alternate, 1 = heterozygous, 2 = homozgyous reference."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "c3fa0da1-2b17-4302-8466-c8da65259c6d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsID</th>\n",
       "      <th>gt</th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1201559</td>\n",
       "      <td>0</td>\n",
       "      <td>1518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs1201559</td>\n",
       "      <td>1</td>\n",
       "      <td>1654</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs1201559</td>\n",
       "      <td>2</td>\n",
       "      <td>1873</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>0</td>\n",
       "      <td>1549</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>1</td>\n",
       "      <td>1652</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>2</td>\n",
       "      <td>1844</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        rsID  gt     N\n",
       "0  rs1201559   0  1518\n",
       "1  rs1201559   1  1654\n",
       "2  rs1201559   2  1873\n",
       "3  rs1783634   0  1549\n",
       "4  rs1783634   1  1652\n",
       "5  rs1783634   2  1844"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_coverage_rs1790218_proxy_variant_genotypes_summary = full_coverage_rs1790218_proxy_variant_genotypes.groupby(['rsID','gt']).size().to_frame('N').reset_index()\n",
    "full_coverage_rs1790218_proxy_variant_genotypes_summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "fd0f33f7-5574-4f91-b20f-67e0824b476a",
   "metadata": {},
   "outputs": [],
   "source": [
    "def calculate_allele_frequencies(homozygous_reference_count, heterozygous_count, homozygous_alternate_count):\n",
    "    total_count = homozygous_reference_count + heterozygous_count + homozygous_alternate_count\n",
    "    \n",
    "    # Calculate the allele frequencies\n",
    "    reference_allele_frequency = (2 * homozygous_reference_count + heterozygous_count) / (2 * total_count) * 100\n",
    "    alternate_allele_frequency = (2 * homozygous_alternate_count + heterozygous_count) / (2 * total_count) * 100\n",
    "    \n",
    "    return reference_allele_frequency, alternate_allele_frequency"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "5271727f-c096-4e94-8db2-a175904b9887",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(53.5183349851338, 46.4816650148662)"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "calculate_allele_frequencies(1873,1654,1518)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "bc950462-4815-4d33-8f88-ea7113ff2421",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(52.92368681863231, 47.07631318136769)"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "calculate_allele_frequencies(1844,1652,1549)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6d4932e8-51ef-45da-be83-1d4b9c6493f4",
   "metadata": {},
   "source": [
    "Let's also do a quick test with the 1KG samples to ensure allele frequencies are as expected."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "dd401cc9-5e6d-45ba-bd11-3386b46ab8bd",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "Thousand_Genomes_IDs = pd.read_csv('/wynton/group/capra/projects/SLC22A10/data/1KG_phase_3_sample_IDs.txt', names = ['ID']) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "f3dbc1cc-4237-4a5c-b231-d1de00bc4ac9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsID</th>\n",
       "      <th>sample_ID</th>\n",
       "      <th>sample</th>\n",
       "      <th>date</th>\n",
       "      <th>location</th>\n",
       "      <th>subregion</th>\n",
       "      <th>region</th>\n",
       "      <th>gt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>674</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HG03006</td>\n",
       "      <td>S_Bengali-1.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Bengali.DG</td>\n",
       "      <td>South Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>675</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HG03007</td>\n",
       "      <td>S_Bengali-2.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Bengali.DG</td>\n",
       "      <td>South Asia</td>\n",
       "      <td>Asia</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2822</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HG00126</td>\n",
       "      <td>S_English-1.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>English.DG</td>\n",
       "      <td>Western Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2823</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HG00128</td>\n",
       "      <td>S_English-2.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>English.DG</td>\n",
       "      <td>Western Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2824</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HG02943</td>\n",
       "      <td>S_Esan-2.DG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Esan.DG</td>\n",
       "      <td>Northern Africa</td>\n",
       "      <td>Africa</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           rsID sample_ID          sample  date    location        subregion  \\\n",
       "674   rs1783634   HG03006  S_Bengali-1.DG   0.0  Bengali.DG       South Asia   \n",
       "675   rs1783634   HG03007  S_Bengali-2.DG   0.0  Bengali.DG       South Asia   \n",
       "2822  rs1783634   HG00126  S_English-1.DG   0.0  English.DG   Western Europe   \n",
       "2823  rs1783634   HG00128  S_English-2.DG   0.0  English.DG   Western Europe   \n",
       "2824  rs1783634   HG02943     S_Esan-2.DG   0.0     Esan.DG  Northern Africa   \n",
       "\n",
       "      region  gt  \n",
       "674     Asia   1  \n",
       "675     Asia   1  \n",
       "2822  Europe   1  \n",
       "2823  Europe   1  \n",
       "2824  Africa   2  "
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rs1790218_proxy_variant_AADR_1KG_genotypes = full_coverage_rs1790218_proxy_variant_genotypes[full_coverage_rs1790218_proxy_variant_genotypes['sample_ID'].isin(Thousand_Genomes_IDs['ID'])]\n",
    "rs1790218_proxy_variant_AADR_1KG_genotypes.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "c394789a-e354-4295-9aae-e930cbea5913",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2503"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(rs1790218_proxy_variant_AADR_1KG_genotypes[rs1790218_proxy_variant_AADR_1KG_genotypes['rsID'] == 'rs1201559'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "810722dd-3b4d-4b5e-adf1-df60c9f9216e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2503"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(rs1790218_proxy_variant_AADR_1KG_genotypes[rs1790218_proxy_variant_AADR_1KG_genotypes['rsID'] == 'rs1783634'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "70482dcf-c93f-4b31-b2bf-d452c8079f9a",
   "metadata": {},
   "source": [
    "It looks like one sample from the original 2,504 1KG sample set is missing."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "5a6ca663-2acd-40a9-ba03-3572ecbebfda",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>HG00139</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         ID\n",
       "38  HG00139"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Thousand_Genomes_IDs[~Thousand_Genomes_IDs['ID'].isin(rs1790218_proxy_variant_AADR_1KG_genotypes['sample_ID'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "9adcb957-ce9a-42c6-9350-ef2b52d60bc3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsID</th>\n",
       "      <th>sample_ID</th>\n",
       "      <th>sample</th>\n",
       "      <th>date</th>\n",
       "      <th>location</th>\n",
       "      <th>subregion</th>\n",
       "      <th>region</th>\n",
       "      <th>gt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3311</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>HG00139</td>\n",
       "      <td>HG00139.SG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GBR.SG</td>\n",
       "      <td>Western Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29345</th>\n",
       "      <td>rs1201559</td>\n",
       "      <td>HG00139</td>\n",
       "      <td>HG00139.SG</td>\n",
       "      <td>0.0</td>\n",
       "      <td>GBR.SG</td>\n",
       "      <td>Western Europe</td>\n",
       "      <td>Europe</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            rsID sample_ID      sample  date location       subregion  region  \\\n",
       "3311   rs1783634   HG00139  HG00139.SG   0.0   GBR.SG  Western Europe  Europe   \n",
       "29345  rs1201559   HG00139  HG00139.SG   0.0   GBR.SG  Western Europe  Europe   \n",
       "\n",
       "       gt  \n",
       "3311    0  \n",
       "29345   0  "
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rs1790218_proxy_variant_genotypes[rs1790218_proxy_variant_genotypes['sample_ID'] == 'HG00139']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ed8c9b1d-fceb-460c-86ce-3161ae7fa418",
   "metadata": {},
   "source": [
    "This individual only has SG rather than both DG and SG samples. Let's calculate allele frequencies for the present 2,503 samples."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "0005d8a7-edf6-4f27-87e6-05a9b89194cb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsID</th>\n",
       "      <th>gt</th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1201559</td>\n",
       "      <td>0</td>\n",
       "      <td>509</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs1201559</td>\n",
       "      <td>1</td>\n",
       "      <td>1177</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs1201559</td>\n",
       "      <td>2</td>\n",
       "      <td>817</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>0</td>\n",
       "      <td>507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>1</td>\n",
       "      <td>1175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>rs1783634</td>\n",
       "      <td>2</td>\n",
       "      <td>821</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        rsID  gt     N\n",
       "0  rs1201559   0   509\n",
       "1  rs1201559   1  1177\n",
       "2  rs1201559   2   817\n",
       "3  rs1783634   0   507\n",
       "4  rs1783634   1  1175\n",
       "5  rs1783634   2   821"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rs1790218_proxy_variant_AADR_1KG_genotypes_summary = rs1790218_proxy_variant_AADR_1KG_genotypes.groupby(['rsID','gt']).size().to_frame('N').reset_index()\n",
    "rs1790218_proxy_variant_AADR_1KG_genotypes_summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "b883b280-c8b2-41b2-9f04-1e1909add457",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(56.15261685976828, 43.84738314023172)"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "calculate_allele_frequencies(817,1177,509)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "b17720a5-236b-422a-bf44-0a60cb5d074a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(56.27247303236117, 43.72752696763884)"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "calculate_allele_frequencies(821,1175,507)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5455b6d4-639a-4f0e-a9a4-72cd541f8be6",
   "metadata": {},
   "source": [
    "These frequencies approximate known allele frequencies for rs1201559 from the full 2,504 individual sample - C: 56.21%, T: 43.79% and rs1783634 - A: 56.25%, G: 43.75%. The discrepancy is likely due to the missing individual as well as potential differences in genotype calls between AADR and the original 1KG data. Let's test these hypotheses."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "318a653a-9f7b-43d5-b88d-ea9b6ec3393a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>HG00096</th>\n",
       "      <th>HG00097</th>\n",
       "      <th>HG00099</th>\n",
       "      <th>HG00100</th>\n",
       "      <th>HG00101</th>\n",
       "      <th>HG00102</th>\n",
       "      <th>...</th>\n",
       "      <th>NA21128</th>\n",
       "      <th>NA21129</th>\n",
       "      <th>NA21130</th>\n",
       "      <th>NA21133</th>\n",
       "      <th>NA21135</th>\n",
       "      <th>NA21137</th>\n",
       "      <th>NA21141</th>\n",
       "      <th>NA21142</th>\n",
       "      <th>NA21143</th>\n",
       "      <th>NA21144</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11</td>\n",
       "      <td>63072310</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|0</td>\n",
       "      <td>...</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|0</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>0|0</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>11</td>\n",
       "      <td>63066671</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|0</td>\n",
       "      <td>...</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|1</td>\n",
       "      <td>0|0</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>1|1</td>\n",
       "      <td>0|0</td>\n",
       "      <td>1|0</td>\n",
       "      <td>0|0</td>\n",
       "      <td>0|1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 2508 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   chr       pos ref alt HG00096 HG00097 HG00099 HG00100 HG00101 HG00102  ...  \\\n",
       "0   11  63072310   C   T     1|1     1|1     1|1     1|0     0|1     0|0  ...   \n",
       "1   11  63066671   A   G     1|1     1|1     1|1     1|0     0|1     0|0  ...   \n",
       "\n",
       "  NA21128 NA21129 NA21130 NA21133 NA21135 NA21137 NA21141 NA21142 NA21143  \\\n",
       "0     1|0     0|1     0|0     1|1     1|1     1|1     0|0     1|0     0|0   \n",
       "1     1|0     0|1     0|0     1|1     1|1     1|1     0|0     1|0     0|0   \n",
       "\n",
       "  NA21144  \n",
       "0     0|1  \n",
       "1     0|1  \n",
       "\n",
       "[2 rows x 2508 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rs1790218_proxy_variants_1KG_phase_3_genotypes = pd.read_csv('/wynton/group/capra/projects/SLC22A10/data/rs1790218_proxy_variants_1KG_phase_3_genotypes.txt', sep = '\\t', header = 0)\n",
    "rs1790218_proxy_variants_1KG_phase_3_genotypes.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "45aa8b4d-9cf6-4acc-81fd-081517922b0a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gt</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>818</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       N\n",
       "gt      \n",
       "0    507\n",
       "1   1178\n",
       "2    818"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rs1201559_1KG_phase_3_genotypes = rs1790218_proxy_variants_1KG_phase_3_genotypes.loc[0].drop(['chr','pos','ref','alt','HG00139']).to_frame('gt')\n",
    "rs1201559_1KG_phase_3_genotypes = rs1201559_1KG_phase_3_genotypes.replace({ '0|0':2, '0|1':1, '1|0':1, '1|1':0})\n",
    "rs1201559_1KG_phase_3_genotypes.groupby('gt').size().to_frame('N')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "30a58cfe-8444-4ecf-8ca0-704f86d24111",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gt</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>820</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       N\n",
       "gt      \n",
       "0    507\n",
       "1   1176\n",
       "2    820"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rs1783634_1KG_phase_3_genotypes = rs1790218_proxy_variants_1KG_phase_3_genotypes.loc[1].drop(['chr','pos','ref','alt','HG00139']).to_frame('gt')\n",
    "rs1783634_1KG_phase_3_genotypes = rs1783634_1KG_phase_3_genotypes.replace({ '0|0':2, '0|1':1, '1|0':1, '1|1':0})\n",
    "rs1783634_1KG_phase_3_genotypes.groupby('gt').size().to_frame('N')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "8d021b89-dd70-473e-8c7b-e37a03bdbac3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(56.212544946064725, 43.787455053935275)"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "calculate_allele_frequencies(818,(591+587),507)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "042cdd28-5307-446b-87a9-09d792871aba",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(56.25249700359568, 43.747502996404315)"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "calculate_allele_frequencies(820,(591+585),507)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "283c770a-2166-4fa2-bf65-70a23abe0ed7",
   "metadata": {},
   "source": [
    "There are some differences in the genotype distributions for both proxy variants among 2,503 1KG phase 3 individuals and those same individuals in the AADR. We will proceed with the AADR allele frequencies. Save the dataframe for visualization."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "ab01f86f-0471-417b-a3e9-fdea30ecea3f",
   "metadata": {},
   "outputs": [],
   "source": [
    "full_coverage_rs1790218_proxy_variant_genotypes.to_csv('full_coverage_rs1790218_proxy_variant_genotypes.txt', sep = '\\t', header = True, index = False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
